hack/devtools/rp_dev_helper.sh

```diff
@@ -1,39 +1,471 @@
+
+    # Don't skip deployment creation when skip_deployments was set to "false" 
+    if is_it_boolean $skip_deployments && ! $skip_deployments; then
+        abort "'skip_deployments' was set to 'false'. ❌⏩ Don't skip certs import"
+    fi
+
+    local files
+    local certs
+    local key_vault
+    key_vault="${keyvault_prefix}-svc"
+    log "Check import certificates for the service keyVault ${key_vault}"
+    certs=("rp-mdm" "rp-mdsd" "cluster-mdsd" "dev-arm" "rp-firstparty" "rp-server")
+    files=("secrets/rp-mdm-self-signed.pem" "secrets/rp-mdsd-self-signed.pem" "secrets/cluster-mdsd-self-signed.pem" "secrets/arm.pem" "secrets/firstparty.pem" "secrets/localhost.pem")
+    skip_and_import_certificates ${key_vault} ${certs[@]} ${files[@]} $skip_deployments
+    
+    key_vault="${keyvault_prefix}-gwy"
+    log "Check import certificates for the gateway keyVault ${key_vault}"
+    certs=("gwy-mdm" "gwy-mdsd")
+    files=("secrets/gwy-mdm-self-signed.pem" "secrets/gwy-mdsd-self-signed.pem")
+    skip_and_import_certificates ${key_vault} ${certs[@]} ${files[@]} $skip_deployments
+
+    key_vault="${keyvault_prefix}-por"
+    log "Check import certificates for the portal keyVault ${key_vault}"
+    certs=("portal-server" "portal-client")
+    files=("secrets/localhost.pem" "secrets/portal-client.pem")
+    skip_and_import_certificates ${key_vault} ${certs[@]} ${files[@]} $skip_deployments
+}
+
+clean_rp_dev_env() {
+    err_str="Usage $0 <LOCATION> [LIST_RESOURCE_GROUPS] [LIST_KEYVAULTS]. Please try again"
+    local location=${1?$err_str}
+    local -a list_resource_groups=("${2:-}")
+    local -a list_keyvaults=("${3:-}")
+    log "########## Deleting Dev Env in $location ##########"
+
+    if [[ $# -lt 2 ]]; then
+        log "Info: One input argument was required. Checking AZURE_PREFIX env var for default values"
+        if [[ -z "${AZURE_PREFIX}" ]]; then
+            abort "Error: AZURE_PREFIX environment variable was not set."
+        fi
+        log "Info: AZURE_PREFIX=${AZURE_PREFIX}"
+    fi
+
+    # Convert input strings to arrays
+    eval "rgs=($list_resource_groups)"
+    eval "kvs=($list_keyvaults)"
+
+   if [[ ${#rgs[@]} -eq 0 ]]; then
+        rg_suffixes=("global" "subscription" "gwy-$location" "aro-$location")
+        for suffix in ${rg_suffixes[@]}; do
+            rgs+=("${AZURE_PREFIX}-$suffix")
+        done
+        log "No resource groups were provided. Use default values for list: ${rgs[*]}"
+    fi
+
+     for rg in ${rgs[@]}; do
+        log "########## Delete Resource Group $rg in $location ##########"
+        az group delete --resource-group "$rg" -y
+    done
+
+    if [[ ${#kvs[@]} -eq 0 ]]; then
+        kv_suffixes=("gwy" "por" "svc" "cls")
+        for suffix in ${kv_suffixes[@]}; do
+            kvs+=("${AZURE_PREFIX}-aro-$location-$suffix")
+        done
+        log "No KeyVaults were provided. Use default values for list: ${kvs[*]}"
+    fi
+
+    for kv in ${kvs[@]}; do
```

Comment:
Add shellcheck disables, remove extra spacing.

```suggestion
        # shellcheck disable=2068
        for suffix in ${rg_suffixes[@]}; do
            rgs+=("${AZURE_PREFIX}-$suffix")
        done
        log "No resource groups were provided. Use default values for list: ${rgs[*]}"
    fi

    # shellcheck disable=2068
    for rg in ${rgs[@]}; do
        log "########## Delete Resource Group $rg in $location ##########"
        az group delete --resource-group "$rg" -y
    done

    if [[ ${#kvs[@]} -eq 0 ]]; then
        kv_suffixes=("gwy" "por" "svc" "cls")
    # shellcheck disable=2068
        for suffix in ${kv_suffixes[@]}; do
            kvs+=("${AZURE_PREFIX}-aro-$location-$suffix")
        done
        log "No KeyVaults were provided. Use default values for list: ${kvs[*]}"
    fi

    # shellcheck disable=2068
    for kv in ${kvs[@]}; do
```